sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -451,7 +451,7 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
       val funcIdentifier = identifier.asFunctionIdentifier
       DropFunctionCommand(funcIdentifier.database, funcIdentifier.funcName, ifExists, isTemp)
 
-    case CreateFunctionStatement(nameParts,
+    case CreateFunction(UnresolvedDBObjectName(nameParts, _),
```

Comment:
I think we should follow `CreateNamespace` and match `ResolvedDBObjectName(catalog, name)` with `if isSessionCatalog(catalog)`

Reply:
let's also add a test that for `CREATE FUNCTION a.b.c`, we should throw a proper error saying that creating function is not supported in v2 catalog.

Reply:
seems we have a test here https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala#L2213